entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -522,4 +505,20 @@ public void total(
     responseObserver.onNext(TotalEntitiesResponse.newBuilder().setTotal(total).build());
     responseObserver.onCompleted();
   }
+
+  private ResultSetMetadata buildMetadataForSelections(List<Expression> selections) {
+    return ResultSetMetadata.newBuilder()
+        .addAllColumnMetadata(
+            () ->
+                selections.stream()
+                    .map(Expression::getColumnIdentifier)
+                    .map(
+                        identifier ->
+                            identifier.getAlias().isEmpty()
```

Comment:
Can you add a test that would fail earlier but will pass now?

Reply:
yep just wanted to get it up, will add a test and fix snyk.

Reply:
test added